unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/Common/QuickBarView.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+public interface IQuickBarView
+{
+    event Action<int> OnQuickBarInputTriggered;
+    event Action<int> OnQuickBarObjectSelected;
+    event Action<BaseEventData> OnSceneObjectDropped;
+    event Action<int> OnSetIndexToDrop;
+
+    void ConfigureEventTrigger(int index, EventTriggerType eventType, UnityAction<BaseEventData> call);
+    void OnQuickBarInputTriggedered(int index);
+    void QuickBarObjectSelected(int index);
+    void RemoveEventTrigger(int index, EventTriggerType eventType);
+    void SceneObjectDropped(BaseEventData data);
+    void SetIndexToDrop(int index);
+    void SetTextureToShortcut(int shortcutIndex, Texture texture);
+}
+
+public class QuickBarView : MonoBehaviour, IQuickBarView
+{
+    public event Action<int> OnQuickBarObjectSelected;
+    public event Action<int> OnSetIndexToDrop;
+    public event Action<BaseEventData> OnSceneObjectDropped;
+    public event Action<int> OnQuickBarInputTriggered;
+
+    [SerializeField] internal QuickBarSlot[] shortcutsImgs;
+    [SerializeField] internal Button[] shortcutsButtons;
+    [SerializeField] internal EventTrigger[] shortcutsEventTriggers;
+    [SerializeField] internal InputAction_Trigger quickBar1InputAction;
+    [SerializeField] internal InputAction_Trigger quickBar2InputAction;
+    [SerializeField] internal InputAction_Trigger quickBar3InputAction;
+    [SerializeField] internal InputAction_Trigger quickBar4InputAction;
+    [SerializeField] internal InputAction_Trigger quickBar5InputAction;
+    [SerializeField] internal InputAction_Trigger quickBar6InputAction;
+    [SerializeField] internal InputAction_Trigger quickBar7InputAction;
+    [SerializeField] internal InputAction_Trigger quickBar8InputAction;
+    [SerializeField] internal InputAction_Trigger quickBar9InputAction;
+
+    private const string VIEW_PATH = "Common/QuickBarView";
+
+    internal static QuickBarView Create()
+    {
+        var view = Instantiate(Resources.Load<GameObject>(VIEW_PATH)).GetComponent<QuickBarView>();
+        view.gameObject.name = "_QuickBarView";
+
+        return view;
+    }
+
+    private void Awake()
+    {
+        for (int i = 0; i < shortcutsButtons.Length; i++)
+        {
+            int buttonIndex = i;
+            shortcutsButtons[buttonIndex].onClick.AddListener(() => QuickBarObjectSelected(buttonIndex));
+        }
+
+        for (int i = 0; i < shortcutsEventTriggers.Length; i++)
+        {
+            int triggerIndex = i;
+            ConfigureEventTrigger(triggerIndex, EventTriggerType.Drop, (eventData) =>
+            {
+                SetIndexToDrop(triggerIndex);
+                SceneObjectDropped(eventData);
+            });
+        }
+
+        quickBar1InputAction.OnTriggered += (action) => OnQuickBarInputTriggedered(0);
```

Comment:
does unsubscribing like this `quickBar1InputAction.OnTriggered -= (action) => OnQuickBarInputTriggedered(0);` actually works? I always assume that It doesn't 😱

Reply:
Hmmm... you know what? I though we can do it but, after your comment, I have been doing some tests to confirm it and... effectively... this unsubscription way doesn't work! I'm going to take a look to all places where I have this kind of subscriptions and apply some fix. Thanks for the warning!

Reply:
Finally I had to make the unsubscriptions in the traditional way. Changes done in https://github.com/decentraland/explorer/pull/2103/commits/aceeefafbd4d1a17025b0304ae4bd3b0ace6f91e